dask_expr/tests/test_shuffle.py

```diff
@@ -689,6 +689,15 @@ def test_shuffle_no_assign(df, pdf):
     assert len([x for x in q.walk() if isinstance(x, Assign)]) == 0
 
 
+@pytest.mark.parametrize("func", ["set_index", "sort_values", "shuffle"])
+def test_respect_context_shuffle(df, pdf, func):
+    with dask.config.set({"dataframe.shuffle.method": "tasks"}):
+        q = getattr(df, func)("x")
+
+    result = q.optimize(fuse=False)
+    assert len([x for x in result.walk() if isinstance(x, TaskShuffle)]) > 0
```

Comment:
nit: We may want to add this for completeness

```suggestion
    with dask.config.set({"dataframe.shuffle.method": "tasks"}):
        q = getattr(df, func)("x")
    result = q.optimize(fuse=False)
    assert len([x for x in result.walk() if isinstance(x, TaskShuffle)]) > 0

    q = getattr(df, func)("x")
    with dask.config.set({"dataframe.shuffle.method": "tasks"}):
        result = q.optimize(fuse=False)
    assert len([x for x in result.walk() if isinstance(x, TaskShuffle)]) > 0

    with dask.config.set({"dataframe.shuffle.method": "p2p"}):
        q = getattr(df, func)("x")
    with dask.config.set({"dataframe.shuffle.method": "tasks"}):
        result = q.optimize(fuse=False)
    assert len([x for x in result.walk() if isinstance(x, P2PShuffle)]) > 0
```